REGRESSION(252640@main): [ macOS iOS15 ] TestWebKitAPI.WebKit.AlternativeServicesDefaultDirectoryCreation is a constant failure

szewai opened the pull request:
#### 3da73044142a7040e9d068489499450e86ce0399
<pre>
REGRESSION(252640@main): [ macOS iOS15 ] TestWebKitAPI.WebKit.AlternativeServicesDefaultDirectoryCreation is a constant failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=242951">https://bugs.webkit.org/show_bug.cgi?id=242951</a>

Reviewed by Chris Dumez.

The path included in the message (i.e. the path that is actually in use) is the real path of temporary folder (i.e.
&quot;/private/var/tmp&quot;, because we resolve paths with realpath() to create sandbox extension), so let&apos;s look for the real
path in the message.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:

Canonical link: <a href="https://commits.webkit.org/252681@main">https://commits.webkit.org/252681@main</a>
</pre>

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:
```diff
@@ -1441,7 +1443,10 @@ - (void)userContentController:(WKUserContentController *)userContentController d
     done = false;
-    EXPECT_TRUE(retrievedString.contains("\"path\": \"/var/tmp\""_s));
+    String tempDirectoryString = tempDirectory;
+    NSString* tempDirectoryInUse = FileSystem::realPath(tempDirectoryString);
```
nit: We can probably call `FileSystem::realPath(tempDirectory)` and get rid of the `tempDirectoryString` local.

cdumez reviewed (approved):
LGTM

szewai committed:
REGRESSION(252640@main): [ macOS iOS15 ] TestWebKitAPI.WebKit.AlternativeServicesDefaultDirectoryCreation is a constant failure
https://bugs.webkit.org/show_bug.cgi?id=242951

Reviewed by Chris Dumez.

The path included in the message (i.e. the path that is actually in use) is the real path of temporary folder (i.e.
"/private/var/tmp", because we resolve paths with realpath() to create sandbox extension), so let's look for the real
path in the message.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:

Canonical link: https://commits.webkit.org/252681@main
